source/postprocess/visualization/thermal_diffusivity.cc

```diff
@@ -0,0 +1,97 @@
+
+
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    namespace VisualizationPostprocessors
+    {
+      template <int dim>
+      ThermalDiffusivity<dim>::
+      ThermalDiffusivity ()
+        :
+        DataPostprocessorScalar<dim> ("thermal_diffusivity",
+                                      update_values | update_q_points )
+      {}
+
+
+
+      template <int dim>
+      void
+      ThermalDiffusivity<dim>::
+      compute_derived_quantities_vector (const std::vector<Vector<double> >              &uh,
+                                         const std::vector<std::vector<Tensor<1,dim> > > &duh,
+                                         const std::vector<std::vector<Tensor<2,dim> > > &,
+                                         const std::vector<Point<dim> > &,
+                                         const std::vector<Point<dim> >                  &evaluation_points,
+                                         std::vector<Vector<double> >                    &computed_quantities) const
+      {
+        const unsigned int n_quadrature_points = uh.size();
+        Assert (computed_quantities.size() == n_quadrature_points,    ExcInternalError());
+        Assert (computed_quantities[0].size() == 1,                   ExcInternalError());
+        Assert (uh[0].size() == this->introspection().n_components,           ExcInternalError());
+
+        MaterialModel::MaterialModelInputs<dim> in(n_quadrature_points,
+                                                   this->n_compositional_fields());
+        MaterialModel::MaterialModelOutputs<dim> out(n_quadrature_points,
+                                                     this->n_compositional_fields());
+
+        in.position = evaluation_points;
+        in.strain_rate.resize(0); // we do not need the viscosity
+        for (unsigned int q=0; q<n_quadrature_points; ++q)
+          {
+            in.pressure[q]=uh[q][this->introspection().component_indices.pressure];
+            in.temperature[q]=uh[q][this->introspection().component_indices.temperature];
+
+            for (unsigned int c=0; c<this->n_compositional_fields(); ++c)
+              in.composition[q][c] = uh[q][this->introspection().component_indices.compositional_fields[c]];
+
+          }
+
+        this->get_material_model().evaluate(in, out);
+
+        for (unsigned int q=0; q<n_quadrature_points; ++q)
+          computed_quantities[q](0) = out.thermal_conductivities[q] / (out.densities[q] * out.specific_heat[q]);
+      }
+    }
+  }
+}
+
+
+// explicit instantiations
+namespace aspect
+{
+  namespace Postprocess
+  {
+    namespace VisualizationPostprocessors
+    {
+      ASPECT_REGISTER_VISUALIZATION_POSTPROCESSOR(ThermalDiffusivity,
+                                                  "thermal diffusivity",
+                                                  "A visualization output object that generates output "
+                                                  "for the thermal diffusivity $\\kappa$.")
```

Comment:
Can you state here that this equals $\frac{k}{c_p \rhp}$ where $k$ is the thermal conductivity? I am perpetually confused by which is which, and assume that others are too.